src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -169,7 +171,19 @@ public int order() {
                 attachSourceFieldContext(request);
             }
             final Set<String> injectedRoles = rolesInjector.injectUserAndRoles(threadContext);
+            boolean isInjectedUser = false;
```

Comment:
Is this flag necessary? `User` already has a flag whether it is injected or not.

Reply:
We want to have to privileges check for the local and after injected user is set. For the rest of the cases, it should go with the normal flow.